src/WallsComputation.cpp

```diff
@@ -129,15 +129,15 @@ void WallsComputation::generateInsets(SliceLayerPart* part)
 
     constexpr float transitioning_angle = 0.5;
 
-    Polygons tubeshape = part->outline.tubeShape(line_width_0 + line_width_x * (inset_count - 1), 0).offset(-10).offset(10);
-    tubeshape.simplify(50, 50);
-    tubeshape.removeColinearEdges(0.03);
-    tubeshape.fixSelfIntersections();
-    tubeshape.removeSmallAreas(INT2MM(line_width_0/2) * INT2MM(line_width_0/2), false); // TODO: complete guess as to when arachne starts breaking, but it doesn't function well when an area is really small apearantly?
-    if (tubeshape.area() > 0)
+    Polygons prepared_outline = part->outline.offset(-10).offset(10);
+    prepared_outline.simplify(50, 50);
+    prepared_outline.removeColinearEdges(0.03);
+    prepared_outline.fixSelfIntersections();
+    prepared_outline.removeSmallAreas(INT2MM(line_width_0/2) * INT2MM(line_width_0/2), false); // TODO: complete guess as to when arachne starts breaking, but it doesn't function well when an area is really small apearantly?
+    if (prepared_outline.area() > 0)
```

Comment:
why not check against `!prepared_outline.empty()` it stands to reason that after all simplifications there are at least three polygons present, which aren't self intersecting and therefore should have an area greater then 0.
This safes computation time, since the area() needs to be calculated while empty just checks well for empty ;-)

Reply:
I'm not sure, but I thought empty is not the same as area 0 in all cases (consider a degenerate polygon of 2 lines in a cycle for instance). You can also have negative area for holes, etc.

Reply:
Just checked, area and empty don't do the same thing. I think this one's better, since we won't have potential infinitesimal extrusions this way (won't fix).

Reply:
I agree with you assessment here.